addons/website/models/website_visitor.py

```diff
@@ -306,11 +306,27 @@ def _link_to_visitor(self, target):
 
         return target
 
-    def _cron_archive_visitors(self):
+    def _cron_clean_inactive_visitors(self):
+        """ Unlink inactive visitors (see '_inactive_visitors_domain' for details).
+
+        Visitors were previously archived but we came to the conclusion that archived visitors
+        have very little value and bloat the database for no reason. """
+
+        inactive_visitors = self.env['website.visitor'].sudo() \
+            .with_context(active_test=False) \
+            .search(self._inactive_visitors_domain())
+        inactive_visitors.unlink()
```

Comment:
Side question: do we let the cron run after migration (aka crash incoming due to millions of visitors) or do we do something at migration time to cleanup the db "like a cron" ?

Reply:
I could simply delete all archived visitors during migration?

We would potentially loose a couple push_tokens in that case, but it's probably acceptable?
(Indeed, I think the first CRON run will just crash unless we help a bit).

Reply:
Yep. But are we doing to try to make a domain in SQL depending on installed modules ?

Reply:
Set as solved